src/prax/parser.cr

```diff
@@ -41,7 +41,13 @@ module Prax
         break if (line = @socket.read_line.chomp).empty?
 
         if match = HEADER_RE.match(line)
-          object.add_header(match[1], match[2])
+          name = match[1]
+          begin
+            value = match[3]
+          rescue
+            value = ""
+          end
+          object.add_header(name, value)
```

Comment:
You may use:

```crystal
name = match[1]
value = match[3]? || ""
```

We expect `match[3]` to exist but it doesn't, so Crystal prevents any further error by raising, otherwise we'd have to deal with a nil value whenever we try to fetch a value from a Array (or Hash), which would be tedious! Alternatively, there is a nilable method, `[]?`, which returns the value if it exists or nil.